File: backend/Kinboard.Api/Models/Job.cs
namespace Kinboard.Api.Models;

public class Job
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? ImageUrl { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    // Recurrence settings (shared by all assignments when UseSharedRecurrence is true)
    public string? Recurrence { get; set; } // RRULE-like: FREQ=DAILY;INTERVAL=3 or FREQ=WEEKLY;INTERVAL=2;BYDAY=MO,TH,SU
    public DateTime? RecurrenceStartDate { get; set; } // Start date for recurrence evaluation
    public DateTime? RecurrenceEndDate { get; set; } // null when indefinite
    public bool RecurrenceIndefinite { get; set; } // true => ignore end date
    
    // Multi-user assignment support
    public bool UseSharedRecurrence { get; set; } = true; // true = all assignments use job's recurrence; false = each assignment has its own
    public ICollection<JobAssignment> Assignments { get; set; } = new List<JobAssignment>();
}

File: kinboard-tv/app/src/main/java/com/kinboard/tv/data/model/Job.kt
package com.kinboard.tv.data.model

import com.google.gson.annotations.SerializedName

data class Job(
    @SerializedName("id")
    val id: Int,
    
    @SerializedName("title")
    val title: String,
    
    @SerializedName("description")
    val description: String? = null,
    
    @SerializedName("imageUrl")
    val imageUrl: String? = null,  // Relative path like "/job-images/job_1.webp"
    
    @SerializedName("createdAt")
    val createdAt: String,  // ISO datetime
    
    @SerializedName("recurrence")
    val recurrence: String? = null,  // RRULE format
    
    @SerializedName("recurrenceStartDate")
    val recurrenceStartDate: String? = null,
    
    @SerializedName("recurrenceEndDate")
    val recurrenceEndDate: String? = null,
    
    @SerializedName("recurrenceIndefinite")
    val recurrenceIndefinite: Boolean? = null,
    
    @SerializedName("useSharedRecurrence")
    val useSharedRecurrence: Boolean? = null,
    
    @SerializedName("assignments")
    val assignments: List<JobAssignment>? = null,
    
    @SerializedName("occurrenceDate")
    val occurrenceDate: String? = null
)
